control/HTTPCacheControl.php

```diff
@@ -80,6 +80,16 @@ public function __construct()
 		}
 	}
 
+	protected function allowChange($level, $force)
+	{
+		$forcingLevel = $level + ($force ? self::LEVEL_FORCED : 0);
+		if ($forcingLevel < $this->forcingLevel) {
+			return false;
+		}
+		$this->forcingLevel = $forcingLevel;
```

Comment:
I understand you're trying to encapsulate repeating functionality here, but:
- The active wording ("allow change") suggests that its a instructive action, not a conditional one
- The way it's used (in an `if` statement) suggests that it's stateless, yet it mutates object state
It's a protected function, so I'm not too worried about the API definition, just struck me as a bit weird. Maybe `setForcingLevel()` returning a boolean is a bit cleaner?

Reply:
My vote would go for `isChangeAllowed`, as `setForcingLevel` implies state change even more than `allowChange`

Reply:
My vote would go for `applyForcingLevel` as the very "apply" implies an implicit changes as part of a piece of business logic whereas "set" feels more like a direct property setter.

Reply:
How about this.


```php
	/**
	 * Instruct the cache to apply a change with a given level, optionally
	 * modifying it with a force flag to increase priority of this action.
	 *
	 * If the apply level was successful, the change is made and the internal level
	 * threshold is incremented.
	 *
	 * @param int $level Priority of the given change
	 * @param bool $force If usercode has requested this action is forced to a higher priority.
	 * Note: Even if $force is set to true, other higher-priority forced changes can still
	 * cause a change to be rejected if it is below the required threshold.
	 * @return bool True if the given change is accepted, and that the internal
	 * level threshold is updated (if necessary) to the new minimum level.
	 */
	protected function applyChangeLevel($level, $force)
```